pkg/controller/replica.go

```diff
@@ -91,7 +91,7 @@ func (r *ReconcileReplica) Reconcile(ctx context.Context, request reconcile.Requ
 	} else {
 		// queue garbage collection if a primary attachment is being demoted
 		if demotionRequested(azVolumeAttachment) {
-			r.triggerGarbageCollection(azVolumeAttachment.Spec.UnderlyingVolume)
+			r.triggerGarbageCollection(context.Background(), azVolumeAttachment.Spec.UnderlyingVolume)
```

Comment:
You can append `//nolint:contextcheck // Garbage collection is asynchronous; context is not inherited by design` to this line to suppress the warning.